platforms/core-runtime/build-process-services/src/main/java/org/gradle/api/internal/classpath/Module.java

```diff
@@ -16,40 +16,51 @@
 package org.gradle.api.internal.classpath;
 
 import org.gradle.internal.classpath.ClassPath;
+import org.jspecify.annotations.NullMarked;
+import org.jspecify.annotations.Nullable;
 
-import java.util.Set;
+import java.util.List;
 
 /**
- * Meta-data about a dynamically loadable module.
+ * Metadata about a module of a Gradle distribution.
  */
+@NullMarked
 public interface Module {
-    /**
-     * Returns the classpath for the module implementation. This is the classpath of the module itself. Does not include any dependencies.
-     */
-    ClassPath getImplementationClasspath();
 
     /**
-     * Returns the classpath containing the runtime dependencies of the module. Does not include any other modules.
+     * Get the name of this module.
      */
-    ClassPath getRuntimeClasspath();
+    String getName();
 
     /**
-     * Returns implementation + runtime.
+     * Returns the classpath for the module implementation. This is the classpath of the module itself. Does not include any dependencies.
      */
-    ClassPath getClasspath();
+    ClassPath getImplementationClasspath();
 
     /**
-     * Returns the modules required by this module.
+     * Gets the names of all modules that this module depends on.
      */
-    Set<Module> getRequiredModules();
+    List<String> getDependencyNames();
 
     /**
-     * Returns the transitive closure of all modules required by this module, including the module itself.
+     * An optional set of Maven module coordinates that identifies this module outside
+     * a Gradle distribution. These are the coordinates that would traditionally be used
+     * to resolve this module from a repository if this module were to live in an external
+     * repository.
      */
-    Set<Module> getAllRequiredModules();
+    @Nullable ModuleAlias getAlias();
```

Comment:
TODO: Add unit tests for getAlias. We don't use alias in this PR but the follow-up PR will use it.